common/reactive/src/main/java/io/helidon/common/reactive/StreamValidationUtils.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c)  2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.helidon.common.reactive;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+/**
+ * Helper methods for stream validation.
+ */
+public class StreamValidationUtils {
+
+    private StreamValidationUtils() {
+    }
+
+    /**
+     * Validation of Reactive Streams Specification for JVM rule 3.3.
+     * <br>
+     * {@code Subscription.request} MUST place an upper bound on possible synchronous
+     * recursion between {@code Publisher} and {@code Subscriber}.
+     *
+     * @param maxDepth   maximal expected recursion depth
+     * @param onExceeded called if recursion is deeper than maxDepth,
+     *                   provided with actual depth and spec compliant exception.
+     * @param <T>        payload type of the subscriber
+     * @return true if valid
+     * @see <a href="https://github.com/reactive-streams/reactive-streams-jvm#3.3">reactive-streams/reactive-streams-jvm#3.3</a>
+     */
+    public static <T> boolean checkRecursionDepth303(int maxDepth, BiConsumer<Long, Throwable> onExceeded) {
```

Comment:
The number in name is unusual. As this is not a test class but a production class, please change the names (in all methods in this class).